build_bundle.py

```diff
@@ -0,0 +1,367 @@
+            format(self.__repo_path, rootfs_tar, self.__branch, component))
+
+
+MKSDCARD_IMAGE_PATH = "meta-xt-prod-aos/doc"
+
+
+class ImageBuilder:
+    def __init__(self, cfg):
+        print("image_builder")
+        self.__cfg = cfg
+        self.__temp = tempfile.mkdtemp()
+        self.__mksdcard_path = os.path.join(
+            self.__cfg.get(SECT_PATH, OPT_BASE), MKSDCARD_IMAGE_PATH)
+        self.__loop_dev = ""
+
+    def cleanup(self):
+        if self.__loop_dev:
+            os.system("sudo losetup -d {}".format(self.__loop_dev))
+        shutil.rmtree(self.__temp, ignore_errors=True)
+
+    def build_image(self, deploy_path):
+        print("building image to {}".format(deploy_path))
+        os.system("{}/mk_sdcard_image.sh -p {} -d {}/image.img -c aos".format(
+            self.__mksdcard_path, deploy_path, self.__temp))
+
+    def connect_image(self):
+        print("connect image")
+        self.__loop_dev = subprocess.check_output(
+            "sudo losetup --find --partscan --show {}/image.img".format(
+                self.__temp),
+            shell=True, encoding='utf-8')
+        print(self.__loop_dev)
+        self.__loop_dev = self.__loop_dev.replace('\n', '').replace('\r', '')
+        print("Got loop dev = {}".format(self.__loop_dev))
+
+    def get_partition_device(self, partno):
+        print("get part device {}".format(partno))
+        if not self.__loop_dev:
+            raise Exception("no loop device found")
+        return "{}p{}".format(self.__loop_dev, partno)
+
+
+DEPLOY_DIR = "build/deploy"
+OPT_PARTITION_NO = "partition_no"
+OPT_DOM_DEPLOY_DIR = "deploy_dir"
+
+
+def get_full_rootfs(dom_update, item, bundle_path, cfg, image_builder):
+    dom_update_gz = "{}.gz".format(dom_update)
+    print("Getting rootfs for {}".format(item))
+    dom_partition = cfg.get(item, OPT_PARTITION_NO)
+    dom_dev = image_builder.get_partition_device(dom_partition)
+
+    #TODO handle situation when user doesn't have sudo.
+    os.system('sudo dd if={} of={}/{} bs=10M'.format(dom_dev, bundle_path,
+                                                     dom_update))
+    os.system('sudo gzip --fast -k {}/{}'.format(bundle_path, dom_update))
+    os.system('sudo rm -rf {}'.format(os.path.join(bundle_path, dom_update)))
+
+    return dom_update_gz
+
+
+def build_rootfs(item, bundle_path, bundle_type, base_dir, cfg, image_builder):
+    print("get_rootfs path = {} bundle_type = {}".format(
+        bundle_path, bundle_type))
+
+    deploy_dir = os.path.join(base_dir, DEPLOY_DIR)
+    print("deploy_dir = {}".format(deploy_dir))
+    component_root = ""
+    search_item = cfg.get(item, OPT_DOM_DEPLOY_DIR)
+    for name in os.listdir(deploy_dir):
+        print("item {}".format(name))
+        if search_item in name:
+            component_root = name
+            break
+
+    print("found {}".format(component_root))
+    if not component_root:
+        raise Exception("component not found")
```

Comment:
Don't use the `Exception` class. Use appropriate exception class (for example, own class or `RuntimeError`)